osu.Game/Screens/Select/FilterCriteria.cs

```diff
@@ -133,10 +153,23 @@ public bool Matches(string value)
                 if (string.IsNullOrEmpty(value))
                     return false;
 
+                if (Exact)
+                    return Regex.IsMatch(value, $@"(^|\s){searchTerm}($|\s)", RegexOptions.IgnoreCase | RegexOptions.CultureInvariant);
```

Comment:
Why is this not just

```suggestion
                    return value.Equals(searchTerm, StringComparison.OrdinalIgnoreCase);
```

What's the whitespace-instead-of-string-{start,end} doing here?

Reply:
because you don't want "you" to match "your"

Reply:
Okay, so it won't match "your", but it might match "will you", or "you will", or whatever, because what it's trying to do is trying to test full word phrases rather than substrings.

Again, doesn't fix the problem with song with a title of just "you", as typing `"you"` will still match a bunch of things with the word `"you"` in it. It'll be some results less but I imagine still a lot.